api/cloudcontroller/ccv3/paginate.go

```diff
@@ -69,3 +64,25 @@ func (requester RealRequester) wrapFirstPage(request *cloudcontroller.Request, o
 
 	return wrapper, warnings, nil
 }
+
+func (requester RealRequester) bulkRetrieval(request *cloudcontroller.Request, obj interface{}, appendToExternalList func(interface{}) error) (IncludedResources, Warnings, error) {
```

Comment:
We could but I am trying to make it more generic and only apply to the commands that we want

Reply:
I'm probably not understanding, what is the difference here between the new `bulkRetrieval` function and and an specific actor setting `MaxPerPage` like [here](https://github.com/cloudfoundry/cli/blob/dd2e81236dc26636657dd7c1cdc47e653d7b8e11/actor/v7action/marketplace.go#L15) and calling the existing `paginate` function?

Reply:
Now I see what you mean.
In the end, I do not have a strong preference here it felt more natural to abstract this into the requests than to have this logic scattered around every action, but I also see the argument of reusability of the paginate function, specially because of the strange interface of that function that ignores outputs of the requests......

If you want I can make the change as you mentioned.